langchain4j-azure-open-ai/src/test/java/dev/langchain4j/model/azure/common/AzureOpenAiChatModelIT.java

```diff
@@ -0,0 +1,96 @@
+package dev.langchain4j.model.azure.common;
+
+import dev.langchain4j.model.azure.AzureOpenAiChatModel;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.chat.common.AbstractChatModelIT;
+import org.junit.jupiter.api.condition.DisabledIf;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.List;
+
+class AzureOpenAiChatModelIT extends AbstractChatModelIT {
+
+    static final AzureOpenAiChatModel AZURE_OPEN_AI_CHAT_MODEL = AzureOpenAiChatModel.builder()
+            .endpoint(System.getenv("AZURE_OPENAI_ENDPOINT"))
+            .apiKey(System.getenv("AZURE_OPENAI_KEY"))
+            .deploymentName("gpt-4o-mini")
+            .build();
+
+    @Override
+    protected List<ChatLanguageModel> models() {
+        return List.of(AZURE_OPEN_AI_CHAT_MODEL);
+    }
+
+    @Override
+    protected String modelName() {
+        return "gpt-4o";
+    }
+
+    @Override
+    protected boolean supportsModelNameParameter() {
+        return false; // TODO implement
+    }
```

Comment:
this should be in unit test, it is testable without API tokens

Reply:
How would you test it now in a unit test?

Reply:
> How would you test it now in a unit test?

I am thinking about mocking http clients. It does not completely replace integration tests.

Reply:
> I am also not sure iabout fragmentizing compatibility tests, this will require all modules to implement more tests and there is a risk some will be forgotten

If conformanse tests will have all the usecases, then not implementing the use case should cause failure, unless this model capability is disabled. Test should check it in the first place

Reply:
> If conformanse tests will have all the usecases, then not implementing the use case should cause failure, unless this model capability is disabled. Test should check it in the first place

This is exactly how it works. My concern is different: if we split compatibility tests into IT and unit tests, it will most probably mean that they will live in separate classes, which might lead to some integration forgetting extending one of the classes. So better to keep all tests in one place.

> I am thinking about mocking http clients. It does not completely replace integration tests.

This should be easy to do once https://github.com/langchain4j/langchain4j/pull/1103 is merged (I plan to address this next after this PR). But this will work only when we do not use the official SDK. For those cases when we use the official SDK (e.g. Azure, AWS, Google), we need to see if mocking their APIs would work.